CoreWiki/Helpers/UrlHelpers.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Text;
+
+namespace CoreWiki.Helpers
+{
+    public class UrlHelpers
+    {
+
+        /// https://stackoverflow.com/questions/25259/how-does-stack-overflow-generate-its-seo-friendly-urls
+        /// <summary>
+        /// Produces optional, URL-friendly version of a title, "like-this-one". 
+        /// hand-tuned for speed, reflects performance refactoring contributed
+        /// by John Gietzen (user otac0n) 
+        /// </summary>
+        public static string URLFriendly(string title)
+        {
+            if (title == null) return "";
+
+            const int maxlen = 80;
+            int len = title.Length;
+            bool prevdash = false;
+            var sb = new StringBuilder(len);
+            char c;
+
+            for (int i = 0; i < len; i++)
+            {
+                c = title[i];
+                if ((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9'))
```

Comment:
I would rather not loop through a string, but rather use a RegularExpression to inspect and insert spaces or hyphens appropriately

Reply:
There are also lots of good suggestions on this stack overflow post where the original code came from: https://stackoverflow.com/questions/25259/how-does-stack-overflow-generate-its-seo-friendly-urls